misc/python/materialize/optbench/util.py

```diff
@@ -8,22 +8,37 @@
 # by the Apache License, Version 2.0.
 
 from pathlib import Path
-from typing import cast
+from re import match
+from typing import Callable, Dict, Optional, cast
 
 import numpy as np
 
 from . import Scenario
 
 
-def str_to_ns(time: str) -> np.timedelta64:
-    """Parses a time format `hh:mm:ss.up_to_9_digits` to a `np.timedelta64`."""
-    h, m, s = time.split(":")
-    s, ns = s.split(".")
-    ns = ns.ljust(9, "0")
-    ns = map(
-        lambda t, unit: np.timedelta64(t, unit), [h, m, s, ns], ["h", "m", "s", "ns"]
+def duration_to_timedelta(duration: str) -> Optional[np.timedelta64]:
+    """Converts a duration like `{time}.{frac}{unit}` to a `np.timedelta64`."""
+
+    frac_to_ns = cast(
+        Dict[str, Callable[[str], str]],
```

Comment:
Is the `cast` needed? I assumed just annotating the type would work:
```
frac_to_ns: Dict[str, Callable[[str], str]] = { ... }
```

Reply:
I didn't realize that I can annotate a local variable.

Reply:
You can! The benefit is that mypy can check that the types do indeed match. Whereas I think mypy interprets a `cast` as "just trust me".